_episodes/07-collaboration.md

```diff
@@ -0,0 +1,183 @@
+you are searching for how to plot a scatter plot with transparency given by
+another column in your dataframe.  An answer may suggest matplotlib, seaborn,
+or another plotting library.  When you integrate the answer into your code,
+you will have to modify the answer to fit your variables and when finally put
+in place, the one line could have come directly from the documentation.  Ideally
+you would use the suggested answer to further research the documentation and
+develop a distinct call.  In this case, you are using an internet forum as a
+shortcut to library documentation.  You don't have to disclose where the original
+idea came from, but for your own benefit it is a good idea to include the URL
+in a comment.
+
+But what if you are searching for something like a binary search written in python,
+or the answer contains a function snippet for performing the task you need?
+Fair use does not use length as a factor, if you directly copy and paste code
+and you want to distribute that code, it would then fall under the original license,
+for stackoverflow, that is CC-BY-SA.
+
+> ## Pop Quiz
+>
+> If you use CC-BY-SA work in your project, what kind of license should you use?
+>
+{: .challenge}
+
+## LLMs and code assistants
+
+A recent development for software engineers is the rise of large language models
+capable of producing code from English descriptions.  The utilities are integrated
+in many search engines, IDEs, or as standalone assistants.  While performance
+and utility can vary wildly, LLMs can increase developer productivity by removing
+some of the tedious jobs.  Just be wary, correct-seeming code can be worse than
+something clearly wrong!
+
+Just as the technological limits of LLMs are still being discovered, the legal
+aspect of AI and licenses are actively being determined in court.  There are three
+phases where licensing and copyright concerns appear in utilizing LLMs:
+
+1. During training, what code has been ingested?  How do the licenses of the
+code affect what is produced?
+2. If you want to refine an LLM, what license is the model distributed with?
+3. If I use code from an LLM, what attribution does it need and will it affect
+my license?
+
+### Ingested code
+
+Unless you are training your own LLM, this is more of an interesting case study
+in copyright than a day-to-day concern.  Since the implementation details of
+many LLMs are proprietary, you may not know what code a model was trained on,
+what license the code used, or if permission was attained to use the software
+for training.
+
+Legal challenges against AI companies have been brought up by artists and
+authors, who allege the generation of verbatim passages or replication of
+artistic style indicate training data included copyrighted material against
+the creators wishes.  Code generation hasn't been included in these lawsuits
+so far, but rulings on other domains could affect how LLMs are trained.
+
```

Comment:
If you are concerned about the origin of the code used to train the LLM, look for LLMs that provide information about their training set.